### DISCUSSION TITLE
スマホ版検索結果リストのページ移動ボタン

### DISCUSSION THREAD

**Discussion by tsubolabo** [increments/qiita-discussions#305] [要望・アイデア・不具合]
スマホでの記事検索についての問題提起です。検索結果が表示されているページの下の方にあるページ移動用ボタンが操作しにくいと感じます。

　「>」ボタンと「>>」ボタンの間隔、「<<」ボタンと「<」ボタンの間隔が狭く、前後のページに移動するつもりが、指の位置が少々ずれただけで、最初や最後のページに移動してしまうことが多々あります。

　こうなると、もう、元の位置まで戻ってやり直す気力も起こりません。ボタンの間隔をもっと広げていただけないでしょうか？

**Comment by tanaka-asami415** [comment]
@tsubolabo 
ご意見ありがとうございます。
前回いただいたご意見 https://github.com/increments/qiita-discussions/discussions/296 と合わせて調査・検討いたします🙇
修正されるまではご不便をおかけしますが、ブラウザの「戻る」操作でご対応いただけますと幸いです。

**Reply by tsubolabo** [reply]
今日の昼ころから、検索結果のページ移動用ボタンの様式が変わりました。
私のこの提案を反映した結果だとすると、私の思いからはかなり外れた結果になっています。

私は今はパソコンで閲覧しています。パソコンでは以前は、
<<　<　1　2　3　...　>　>>
のように表示されていたと記憶していますが、現在は
<　1/38　>
のように変わっています。これでは非常に不便です。最初や最後、途中のページへ一気に飛ぶことができなくなりました。

今、スマホでも確認しましたが、パソコンと同じ
<　1/38　>
という表示です。間違えて<<や>>を押す危険はなくなりましたが、私が期待していたものとは違います。私の希望は、

> （**<<と<、>と>>**）ボタンの間隔をもっと広げていただけないでしょうか？

でした。

**Reply by tsubolabo** [reply]
今回の変更により、スマホでの操作性は以前より良くなっていますが、スマホ版の仕様を変えることにより、パソコン版の画面まで道連れになって使いにくくなるようであれば、他の方からの苦情がありそうです。両画面での折り合いがつかないのであれば、どちらも以前の仕様のままで結構です。

**Reply by tanaka-asami415** [reply]
@tsubolabo 
別の目的で検索ページのアップデートを行ったのですが、それに伴って一部のインターフェースも変わっております。
検索の操作性については今後対応していきますので、お待ちいただけますと幸いです🙇

**Reply by tsubolabo** [reply]
そうでしたか。私の無理なお願いが原因かと思い心配しました。
>別の目的で検索ページのアップデートを行ったのですが、それに伴って一部のインターフェースも変わっております。

それでしたら、ひとつ気になることがあります。
数日前に試したときには、
created:>=2022-09-10 created:<=2022-09-12
のように検索したとき、9/10～9/12 に投稿した記事だけが正しく拾い出せていたはず（思い違いかもしれません(^_^;)）なのですが、今日やってみたところ、9/9 の記事まで拾っています（件数からみて9/9の全部ではなさそうです）

また、検索結果のソート順の指定は、数日前までは「いいね順」だったと思いますが、今は以前の呼び方の「LGTM順」になっています。

余談ですが、先日下記の記事を投稿し、現在、内容拡充のために検索機能を色々と試しているところです。少々僭越ではありますが希望的なことも書いておりますので、操作性改善の参考になれば幸いです。なお、昨日から「>>」ボタンが使えなくなって機能のテストにも少々苦労しております。
[【Qiita検索オプション】試行錯誤と応用技](https://qiita.com/tsubolabo/items/5ab55372701bffff8103)

**Reply by tanaka-asami415** [reply]
@tsubolabo 
ご報告ありがとうございます🙏

> 数日前に試したときには、
created:>=2022-09-10 created:<=2022-09-12
のように検索したとき、9/10～9/12 に投稿した記事だけが正しく拾い出せていたはず（思い違いかもしれません(^_^;)）なのですが、今日やってみたところ、9/9 の記事まで拾っています（件数からみて9/9の全部ではなさそうです）

こちら不具合ですので修正いたします！🙇

> 検索結果のソート順の指定は、数日前までは「いいね順」だったと思いますが、
今は以前の呼び方の「LGTM順」になっています。

検索ページのアップデートで戻ってしまいました…💦こちらも修正します！

**Reply by mkdk09** [reply]
@tsubolabo
> 数日前に試したときには、
created:>=2022-09-10 created:<=2022-09-12
のように検索したとき、9/10～9/12 に投稿した記事だけが正しく拾い出せていたはず（思い違いかもしれません(^_^;)）なのですが、今日やってみたところ、9/9 の記事まで拾っています（件数からみて9/9の全部ではなさそうです）

こちらの不具合の修正が完了しました！
ぜひご確認ください🙏

> 検索結果のソート順の指定は、数日前までは「いいね順」だったと思いますが、
今は以前の呼び方の「LGTM順」になっています。

こちらいいね数順に修正しました！

**Reply by tsubolabo** [reply]
[mkdk09](https://github.com/mkdk09) さん

ありがとうございます。「いいね数順」は確認できました。
しかし、「指定期間外の記事まで表示される」件については、今の検索画面のままでは改善されたかどうかの確認は困難です。「<, >」ボタンだけの仕様で、100ページも捲るのは無理です。
以前の画面のように、飛び越しスキップできる環境には戻らないのでしょうか？

**Reply by tanaka-asami415** [reply]
@tsubolabo 
ご確認ありがとうございます🙏
>「<, >」ボタンだけの仕様で、100ページも捲るのは無理です。
以前の画面のように、飛び越しスキップできる環境には戻らないのでしょうか？

恐れ入りますが、現時点では戻す予定はございません🙇
また、100ページ以上の検索結果を閲覧することは想定していないため
検索条件を絞り込んで、表示されるページ数を少なくしていただけると幸いです。
それでも難しい場合はアドレスバーに表示れている、クエリパラメーターで代用ください。

例えば、`Qiita`を検索して2ページ目を表示した際、アドレスバーには以下が表示されているかと思います。
`https://qiita.com/search?q=Qiita&sort=&stocked=&page=2`
`page=2`のところを`page=10`など、飛びたいページ数に変更していただければスキップできます。

**Reply by tsubolabo** [reply]
> page=2のところをpage=10など、飛びたいページ数に変更していただければスキップできます。

え～っ。そこまで求めますか？
「数式の下に出るスクロールバーが気になるのなら、CSSを勝手に書き変えてください」というのと同レベルの高度な要求のような気がします。
それに、スマホでは、アドレスバーなど何処にも見つかりませんが・・・。

以前に出来ていてユーザーが便利に使っていたものを、突然、出来なくするという理由が全く理解できません。

> 恐れ入りますが、現時点では戻す予定はございません

そこまで、断言されるのであれば、これ以上お願いするのは無理でしょう。納得できないまま諦めます。

**Reply by yoshinoyuya** [reply]
@tsubolabo さん率直な疑問点なのですが
> それに、スマホでは、アドレスバーなど何処にも見つかりませんが・・・。

こちらに関して、Android,iOS,ブラウザ関係なくスマホでアドレスバーを確認できると思っているのですが
どのような環境で表示されないのでしょうか

**Reply by tsubolabo** [reply]
@yoshinoyuya 

　言葉での表現が難しいので、スマホのスクリーンコピーを添付します。最初の頃はブラウザで閲覧していたのですが、あるとき、訳が分からないままこのアプリがインストールされてしまいまして、今はこれで閲覧しています。ブラウザのようなURL入力欄はありません。

<img src="https://user-images.githubusercontent.com/42261605/202365170-47ea6980-d03b-45eb-a5eb-670acef819ea.png" width="200px"><img src="https://user-images.githubusercontent.com/42261605/202365216-bbe94f53-958a-45b3-8740-720fa2d8de1f.png" width="200px">

**Reply by yoshinoyuya** [reply]
@tsubolabo さん
教えて頂きありがとうございます。確かにアプリでの閲覧の場合は、アドレスバーはなさそうですね…

@tanaka-asami415 さん
提案になるのですが、検索オプションに降順を追加するのはどうでしょうか
ある程度の需要もありそうですし、今回の要望に対して1つの解決策ともなるかと思います。

**Reply by tanaka-asami415** [reply]
@tsubolabo 
スクリーンショットの添付ありがとうございます🙇
ご状況理解いたしました。

@yoshinoyuya  
代替案をいただきありがとうございます！🙏
確かにです！検索機能のご要望は他にも色々といただいており、合わせて検討したいと思います💪

**Comment by tanaka-asami415** [comment]
@tsubolabo
> 以前の画面のように、飛び越しスキップできる環境には戻らないのでしょうか？

上記、お待たせしました。
以前、直近では変更を予定していなかったためクローズとなっておりましたが、最後のページへ遷移するボタンと最初のページへ遷移するボタンを追加しましたので、ご報告します！
アクセシビリティに考慮して、クリック範囲は広めになっているので、以前よりタップもしやすやすくなっていると思います。
https://qiita.com/release-notes/2024-07-25
